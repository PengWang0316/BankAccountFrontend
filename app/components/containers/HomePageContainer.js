import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withAuthenticator } from 'aws-amplify-react';

import { amplifyAuthSignOption } from '../../config';
import { currentAuthenticatedUser as currentAuthenticatedUserAction } from '../../actions/UserActions';
import UserList from '../UserList';

export const HomePageContainer = ({ user, currentAuthenticatedUser }) => {
  useEffect(() => {
    if (!user) currentAuthenticatedUser();
  });

  return (
    <UserList />
  );
};

HomePageContainer.propTypes = {
  user: PropTypes.objectOf(PropTypes.string),
  currentAuthenticatedUser: PropTypes.func.isRequired,
};
HomePageContainer.defaultProps = { user: null };

/* istanbul ignore next */
const mapStateToProps = state => ({ user: state.user });
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  currentAuthenticatedUser: user => dispatch(currentAuthenticatedUserAction(user)),
});
export default withAuthenticator(
  connect(mapStateToProps, mapDispatchToProps)(HomePageContainer),
  amplifyAuthSignOption,
);
