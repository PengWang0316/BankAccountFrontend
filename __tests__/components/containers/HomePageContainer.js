import React from 'react';
import renderer from 'react-test-renderer';

import { HomePageContainer } from '../../../app/components/containers/HomePageContainer';

jest.mock('../../../app/components/UserList', () => 'UserList');

describe('HomePageContainer', () => {
  const defaultProps = {
    currentAuthenticatedUser: jest.fn(),
    user: { id: 'id' },
  };

  beforeEach(() => jest.clearAllMocks());

  test('Call with user', () => {
    renderer.create(<HomePageContainer {...defaultProps} />);
    expect(defaultProps.currentAuthenticatedUser).not.toHaveBeenCalled();
  });

  test('Call without user', () => {
    renderer.create(<HomePageContainer {...{ ...defaultProps, user: null }} />);
    expect(defaultProps.currentAuthenticatedUser).toHaveBeenCalledTimes(1);
  });

  test('HomePageContainer with a user snapshot', () => expect(renderer.create(<HomePageContainer {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('HomePageContainer without a user snapshot', () => expect(renderer.create(<HomePageContainer {...{ ...defaultProps, user: null }} />).toJSON()).toMatchSnapshot());
});
