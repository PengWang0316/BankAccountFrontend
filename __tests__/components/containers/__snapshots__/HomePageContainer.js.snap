// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContainer HomePageContainer with a user snapshot 1`] = `<UserList />`;

exports[`HomePageContainer HomePageContainer without a user snapshot 1`] = `<UserList />`;
